lib/std/special/compiler_rt/fixXfYi.zig

```diff
@@ -0,0 +1,224 @@
+const std = @import("std");
+const math = std.math;
+const Log2Int = math.Log2Int;
+const is_test = @import("builtin").is_test;
+
+pub inline fn fixXfYi(comptime I: type, a: anytype) I {
+    @setRuntimeSafety(is_test);
+
+    const F = @TypeOf(a);
+    const float_bits = @typeInfo(F).Float.bits;
+    const int_bits = @typeInfo(I).Int.bits;
+    const rep_t = std.meta.Int(.unsigned, float_bits);
+    const sig_bits = math.floatMantissaBits(F);
+    const exp_bits = math.floatExponentBits(F);
+    const fractional_sig_bits = sig_bits - @boolToInt(F == f80);
```

Comment:
There's an API for this condition as of 5b8ac9821dd25c3e5282130b4d93d6c5b7debb08

https://github.com/ziglang/zig/blob/17631cb2d30bf2d7b10401cf8f784a599bade5c5/lib/std/math/float.zig#L46

Reply:
Oh excellent - I'll use this to make the intent clearer, thanks

The `floatMantissaDigits`/`floatMantissaBits` naming is a bit confusing, isn't it? I wonder if `floatFractionalMantissaBits` would be a better function to provide, even if it is long-winded

Reply:
I carried it over from what it's called in other languages but I was thinking too after posting this that maybe it could have a better name, I was thinking `floatFractionalBits` to match the rest of them (returning 1 less than it does right now).

Reply:
Perfect. Thanks for the input, I think the new name is much clearer